FirebaseFunctions/Tests/Unit/FunctionsSerializerTests.swift

```diff
@@ -213,12 +239,66 @@ class FunctionsSerializerTests: XCTestCase {
     XCTAssertEqual(expected, try serializer.encode(input) as? NSDictionary)
   }
 
-  func testDecodeMap() {
+  func testEncodeDictionaryWithInvalidElements() {
+    let input = ["TEST_CustomObj": CustomObject()] as NSDictionary
+
+    XCTAssertThrowsError(try serializer.encode(input)) { error in
+      guard case let .unsupportedType(typeName: typeName) = error as? FunctionsSerializer
+        .Error else {
+        return XCTFail("Unexpected error: \(error)")
+      }
+
+      XCTAssertEqual(typeName, "CustomObject")
+    }
+  }
+
+  func testEncodeDictionaryWithInvalidNestedDictionary() {
+    let input =
+      ["TEST_NestedDict": ["TEST_CustomObj": CustomObject()] as NSDictionary] as NSDictionary
+
+    XCTAssertThrowsError(try serializer.encode(input)) { error in
+      guard case let .unsupportedType(typeName: typeName) = error as? FunctionsSerializer
+        .Error else {
+        return XCTFail("Unexpected error: \(error)")
+      }
+
+      XCTAssertEqual(typeName, "CustomObject")
+    }
+  }
+
+  func testDecodeDictionary() throws {
     let input = ["foo": 1, "bar": "hello", "baz": [3, 9_876_543_210]] as NSDictionary
     let expected = ["foo": 1, "bar": "hello", "baz": [3, 9_876_543_210]] as NSDictionary
     XCTAssertEqual(expected, try serializer.decode(input) as? NSDictionary)
   }
 
+  func testDecodeDictionaryWithInvalidElements() {
+    let input = ["TEST_CustomObj": CustomObject()] as NSDictionary
+
+    XCTAssertThrowsError(try serializer.decode(input)) { error in
+      guard case let .unsupportedType(typeName: typeName) = error as? FunctionsSerializer
+        .Error else {
+        return XCTFail("Unexpected error: \(error)")
+      }
+
+      XCTAssertEqual(typeName, "CustomObject")
+    }
```

Comment:
Nice! This block is used a good bit. WDYT about refactoring this block out to a helper function with a signature like?

```swift
func assertThrowsUnsupportedTypeError<T>(
    _ expression: @autoclosure () throws -> T,
    typeName: String,
)
```

```swift
assertThrowsUnsupportedTypeError(try serializer.decode(input), typeName: "CustomObject")
```

Reply:
I wasn’t quite sure initially but then found a way to make Xcode highlight the correct line (and not the helper method) when a test fails:

```swift
func assert<T>(
    _ expression: @autoclosure () throws -> T,
    throwsUnsupportedTypeErrorWithName expectedTypeName: String,
    line: UInt = #line
)
```

(Naming is weird, of course, and I don’t really like either option.)